src/WebJobs.Script.WebHost/Configuration/ScriptApplicationHostOptionsSetup.cs

```diff
@@ -61,6 +69,69 @@ public void Configure(string name, ScriptApplicationHostOptions options)
                 options.IsSelfHost = options.IsSelfHost;
                 options.IsStandbyConfiguration = true;
             }
+
+            options.AreZipDeploymentAppSettingsValid = ValidateZipDeploymentAppSettings();
+            options.ScmRunFromPackageBlobExists = BlobExists();
+            options.IsZipDeployment = IsZipDeployment(options);
+            options.IsFileSystemReadOnly = options.IsZipDeployment;
+        }
+
+        private bool BlobExists()
+        {
+            var blobExists = _cloudBlockBlobHelperService.BlobExists(_environment.GetEnvironmentVariable(ScmRunFromPackage), EnvironmentSettingNames.ScmRunFromPackage, _logger).GetAwaiter().GetResult();
+            _logger.LogInformation($"Checked if ${EnvironmentSettingNames.ScmRunFromPackage} points to an existing blob: ${blobExists}");
+            return blobExists;
+        }
+
+        private bool IsZipDeployment(ScriptApplicationHostOptions options)
+        {
+            // If the app is using the new app settings for zip deployment, we don't need to check further, it must be a zip deployment.
+            bool usesNewZipDeployAppSettings = IsValidZipSetting(_environment.GetEnvironmentVariable(AzureWebsiteZipDeployment)) ||
+                IsValidZipSetting(_environment.GetEnvironmentVariable(AzureWebsiteAltZipDeployment)) ||
+                IsValidZipSetting(_environment.GetEnvironmentVariable(AzureWebsiteRunFromPackage));
+
+            if (usesNewZipDeployAppSettings)
+            {
+                return usesNewZipDeployAppSettings;
+            }
+
+            // Check if the old app setting for zip deployment is set, SCM_RUN_FROM_PACKAGE.
+            // If not, this is not a zip deployment. If yes, we still need to check if the blob exists,
+            // since this app setting is always created for Linux Consumption, even when it is not used.
+            // In portal deployment, SCM_RUN_FROM_PACKAGE will be set, but we will be using Azure Files instead.
+            if (!IsValidZipUrl(_environment.GetEnvironmentVariable(ScmRunFromPackage)))
+            {
+                return false;
+            }
+
+            // If SCM_RUN_FROM_PACKAGE is a valid app setting and Azure Files are not being used, it must be a zip deployment.
+            if (string.IsNullOrEmpty(_environment.GetEnvironmentVariable(AzureFilesConnectionString)) &&
```

Comment:
this check seems new. does it matter if AzureFilesConnectionString is set or not when deciding if the app is using ZipDeployment?

Will this work?

bool IsZipDeployment(bool scmRunFromPackageBlobExists)
{
scmRunFromPackageBlobExists = false;
bool runFromPkgConfigured = IsValidZipSetting(_environment.GetEnvironmentVariable(AzureWebsiteZipDeployment)) ||
                IsValidZipSetting(_environment.GetEnvironmentVariable(AzureWebsiteAltZipDeployment)) ||
                IsValidZipSetting(_environment.GetEnvironmentVariable(AzureWebsiteRunFromPackage));

   if (runFromPkgConfigured) {return true;}

bool scmRunFromPkgConfigured = IsValidZipSetting(_environment.GetEnvironmentVariable(ScmRunFromPackage)) && BlobExists(_environment.GetEnvironmentVariable(ScmRunFromPackage);

scmRunFromPackageBlobExists = scmRunFromPkgConfigured;

return scmRunFromPkgConfigured;
}

Reply:
> does it matter if AzureFilesConnectionString is set or not when deciding if the app is using ZipDeployment?

You have a point, it should not matter. My check is weaker than yours; I say it's zip deployment if there's absence of other app settings (Azure Files), I don't even bother checking if the blob exists (which we should, just to be safe).

I'll update the check to be explicit like yours; only if the app setting `SCM_RUN_FROM_PACKAGE` is valid *and* the blob exists, it means we are using zip deployment.